HW2/P3/mandelbrot.pyx

```diff
@@ -4,28 +4,183 @@ cimport cython
+            c_real=AVX.make_float8(r_coords[i,j_s+7],
+                                        r_coords[i,j_s+6],
+                                        r_coords[i,j_s+5],
+                                        r_coords[i,j_s+4],
+                                        r_coords[i,j_s+3],
+                                        r_coords[i,j_s+2],
+                                        r_coords[i,j_s+1],
+                                        r_coords[i,j_s])
+
+            # CREATING THE IMAGINARY COMPONENT OF C BASED ON THE REAL PART COORDINATES IN
+            # AS A FLOAT 8 WITH THE DIFFERENT VALUES IN PARALLEL
+            c_imag=AVX.make_float8(i_coords[i,j_s+7],
+                                        i_coords[i,j_s+6],
+                                        i_coords[i,j_s+5],
+                                        i_coords[i,j_s+4],
+                                        i_coords[i,j_s+3],
+                                        i_coords[i,j_s+2],
+                                        i_coords[i,j_s+1],
+                                        i_coords[i,j_s])
+
+            # DEFINING THE INTIAL VALUES OF Z REAL COMPLEX AND ITERATIONS AS 0
+            z_r=z_0
+            z_i=z_0
+            iter_float8=z_0
+
+            for ii in range(max_iterations):
+                # CREATING THE MASK COMPARING THE MAGNITUDE OF Z TO 2^2 WHICH IS 4. 
+                mask_t=AVX.less_than(AVX.fmadd(z_i, z_i, AVX.mul(z_r,z_r)), z_4)
+
+                # THIS COMPARE USES THE ABOVE TO KNOW IF WE SHOULE EXIT THE CODE 
+                # IF THIS IS TRUE WE WILL EXIT THE CODE, AND IT WILL ONLY BE TRUE IF 
+                # ALL OF THE VALUES IN THE NOT_GO_MASK IS FULL
+                if not AVX.signs(mask_t):
+                    break
+
+                # CREATING FLOAT 8 FOR THE REAL AND UPDATIND THE VALUES
+                # WITH THE ADDED C THROUGH THE EQUATION
+                # Z_R= RE(Z*Z+C)=Z_R^2-Z_I^2+C_R
+                z_r_temp=z_r
+                z_r=AVX.add(AVX.sub(AVX.mul(z_r,z_r),AVX.mul(z_i,z_i)),c_real)
+
+                # CREATING FLOAT 8 FOR THE IMAGINARY PART OF C FOR THE 
+                # EQUATION DESCRIBED ABOVE BUT APPLIED TO IMAGINARY SUCH THAT
+                # Z_I=Z_R*Z_I*2
+                z_i=AVX.fmadd(AVX.mul(z_r_temp,z_i), z_2, c_imag)
+
+                # ONCE WE OBTAIN THE VALUE ABOVE WE CAN ADD THE MASK TO THE ITERATION
+                # AS THE MASK WILL EITHER BE 1 OR 0 DEPENDING ON THE INDEQUALITY 
+                # BOOLEAN RESULT FROM THE PART ABOVE
+                iter_float8 = AVX.add(iter_float8,AVX.bitwise_and(mask_t, z_1))
+            
+            # THIS PORTION TRANSFERS ALL OF THE DATA STORED IN AVX INTO THE OUT_COUNTS ARRAYS
+            # TO BE EXPORTED TO THE OUTSIDE FUNCTION PERFORMING THE COMPUTATION
+            for k in range(8):
+                out_counts[i,j+k]= <np.uint32_t> ((<np.float32_t*> &iter_float8)[k])
```

Comment:
Was there a reason that you didn't use the AVX.to_mem function? As far as I understand, it would perform this loop with a single command if you called it right. It's no great concern as you got it to perform as desired, just curious if this seemed better or not.